cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsAutoScalingService.java

```diff
@@ -143,6 +149,27 @@ private void waitForGroup(AmazonEC2Client amClient, AmazonAutoScalingClient asCl
         }
     }
 
+    private void checkLastScalingActivity(AmazonAutoScalingClient asClient, AmazonAutoScalingRetryClient asRetryClient, String autoScalingGroupName,
+            Date timeBeforeASUpdate) throws AmazonAutoscalingFailed {
+        try {
+            Waiter<DescribeScalingActivitiesRequest> autoscalingActivitesWaiter = customAmazonWaiterProvider
+                    .getAutoscalingActivitesWaiter(asClient, timeBeforeASUpdate);
+            autoscalingActivitesWaiter.run(new WaiterParameters<>(new DescribeScalingActivitiesRequest().withAutoScalingGroupName(autoScalingGroupName)));
+        } catch (Exception e) {
+            LOGGER.error("Failed to list activities: {}", e.getMessage(), e);
+            throw new AmazonAutoscalingFailed(e.getMessage(), e);
+        }
+        DescribeScalingActivitiesResult describeScalingActivitiesResult = asRetryClient
+                .describeScalingActivities(new DescribeScalingActivitiesRequest().withAutoScalingGroupName(autoScalingGroupName));
+        Optional<Activity> firstActivity = describeScalingActivitiesResult.getActivities().stream().findFirst()
```

Comment:
Earlier we had a similar logic and support cases where it turned out that looking for a failed activity is not a good approach. As activities could have gone wrong in the past for a scaling activity which is not related to the actual/latest one. I think we need to check the state of the latest activity only.